testing/spectypes/premine-state.go

```diff
@@ -0,0 +1,419 @@
+package spectypes
+
+import (
+	"context"
+
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/pkg/errors"
+	"github.com/prysmaticlabs/prysm/v3/beacon-chain/core/altair"
+	b "github.com/prysmaticlabs/prysm/v3/beacon-chain/core/blocks"
+	"github.com/prysmaticlabs/prysm/v3/beacon-chain/core/helpers"
+	"github.com/prysmaticlabs/prysm/v3/beacon-chain/state"
+	state_native "github.com/prysmaticlabs/prysm/v3/beacon-chain/state/state-native"
+	"github.com/prysmaticlabs/prysm/v3/beacon-chain/state/stateutil"
+	fieldparams "github.com/prysmaticlabs/prysm/v3/config/fieldparams"
+	"github.com/prysmaticlabs/prysm/v3/config/params"
+	"github.com/prysmaticlabs/prysm/v3/consensus-types/blocks"
+	"github.com/prysmaticlabs/prysm/v3/container/trie"
+	"github.com/prysmaticlabs/prysm/v3/crypto/bls"
+	"github.com/prysmaticlabs/prysm/v3/encoding/bytesutil"
+	enginev1 "github.com/prysmaticlabs/prysm/v3/proto/engine/v1"
+	ethpb "github.com/prysmaticlabs/prysm/v3/proto/prysm/v1alpha1"
+	"github.com/prysmaticlabs/prysm/v3/runtime/interop"
+	"github.com/prysmaticlabs/prysm/v3/runtime/version"
+)
+
+var errUnsupportedVersion = errors.New("schema version not supported by premineGenesisConfig")
+
+type premineGenesisConfig struct {
+	GenesisTime uint64
+	NVals       uint64
+	Version     int          // as in "github.com/prysmaticlabs/prysm/v3/runtime/version"
+	GB          *types.Block // geth genesis block
+}
+
+// NewPreminedGenesis creates a genesis BeaconState at the given fork version, suitable for using as an e2e genesis.
+func NewPreminedGenesis(ctx context.Context, t, nvals uint64, version int, gb *types.Block) (state.BeaconState, error) {
```

Comment:
~~I am a bit surprised by the location of this file. Is there any reason it is in our `spectests` folder ?~~ Just realised its a new package